bin/cron/applab_datasets/spotify

```diff
@@ -121,17 +125,22 @@ def main
   get_token
   fb = FirebaseHelper.new('shared')
 
+  if $token.nil_or_empty?
+    Honeybadger.notify_cronjob_error(error_message: "Did not to retrieve Spotify playlist data due to missing authentication token.",)
+    return
+  end
+
   records, columns = get_playlist_data(CHART_TYPE[:TOP], TOP_50_GLOBAL_ID)
-  fb.upload_live_table('Top 50 Worldwide', records, columns)
+  fb.upload_live_table('Top 50 Worldwide', records, columns) unless records.nil_or_empty?
```

Comment:
should we log to honeybadger if `records.nil_or_empty?`? i don't think we ever expect this to be empty unless something has gone wrong

Reply:
Good call! I added the logging, which made me realize we could clean up the way we're calling the playlists to keep things as dry as possible: https://github.com/code-dot-org/code-dot-org/pull/44291/commits/46e958d0d8a7c4bc58817c82012cfea636b8e140

Been a while since I've worked with ruby so let me know if there's a more ideal, cleaner way to do this!